sdk/samples/iot/paho_iot_pnp_sample.c

```diff
@@ -577,6 +576,37 @@ static void process_device_property_message(
         send_reported_property(
             property_reported_maximum_temperature_name, device_maximum_temperature, -1, confirm);
       }
+
+      // Skip to next property value
+      rc = az_json_reader_next_token(&jr);
+      if (az_result_failed(rc))
+      {
+        IOT_SAMPLE_LOG_ERROR("Could not move to next property name");
+      }
+    }
+    else
+    {
+      IOT_SAMPLE_LOG_AZ_SPAN("Unknown Property Received:", jr.token.slice);
+      // The JSON reader must be advanced regardless of whether the property
+      // is of interest or not.
+      rc = az_result_failed(az_json_reader_next_token(&jr));
+      if (az_result_failed(rc))
+      {
+        IOT_SAMPLE_LOG_ERROR("Could not move to next property name");
+      }
+
+      // Skip children in case the property value is an array
+      rc = az_result_failed(az_json_reader_skip_children(&jr));
+      if (az_result_failed(rc))
+      {
+        IOT_SAMPLE_LOG_ERROR("Could not move to next property name");
+      }
+
+      rc = az_result_failed(az_json_reader_next_token(&jr));
```

Comment:
If we need to call this `next_token` for each case, then we can move it outside the `else` logic